projects/core/src/cart/store/actions/cart.action.ts

```diff
@@ -24,19 +32,44 @@ export const CLEAR_CART = '[Cart] Clear Cart';
 export const DELETE_CART = '[Cart] Delete Cart';
 export const DELETE_CART_FAIL = '[Cart] Delete Cart Fail';
 
-export class CreateCart {
+export class CreateCart extends EntityLoadAction {
   readonly type = CREATE_CART;
-  constructor(public payload: any) {}
-}
-
-export class CreateCartFail {
+  constructor(
+    public payload: {
+      userId: string;
+      tempCartId: string;
+      extraData?: Object;
+      oldCartId?: string;
+      toMergeCartGuid?: string;
+    }
+  ) {
+    super(MULTI_CART_DATA, payload.tempCartId);
+  }
+}
+
+export class CreateCartFail extends EntityFailAction {
   readonly type = CREATE_CART_FAIL;
-  constructor(public payload: any) {}
-}
-
-export class CreateCartSuccess {
+  constructor(
+    public payload: {
+      userId: string;
+      tempCartId: string;
+      error: any;
+      extraData?: Object;
```

Comment:
I looked at the usage and I don't think it will be possible to remove that.
At the moment I typed that for CreateCart actions, but ideally, after cleaning all unused actions I would like to rename that to property `cartType` with value `active` as this is the only property now.